enclave-manager/web/src/emui/enclaves/EnclaveList.tsx

```diff
@@ -0,0 +1,75 @@
+import { Button, ButtonGroup, Flex, Spinner, Tab, TabList, TabPanel, TabPanels, Tabs } from "@chakra-ui/react";
+import { useKurtosisClient } from "../../client/KurtosisClientContext";
+import { useEffect, useState } from "react";
+import { isDefined } from "../../utils";
+import { EnclavesTable } from "../../components/enclaves/EnclavesTable";
+import { FiPlus, FiTrash2 } from "react-icons/fi";
+import { EnclaveFullInfo } from "./types";
+
+export const EnclaveList = () => {
+  const kurtosisClient = useKurtosisClient();
+  const [enclaves, setEnclaves] = useState<EnclaveFullInfo[]>();
+  const [selectedEnclaves, setSelectedEnclaves] = useState<EnclaveFullInfo[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      const enclavesResponse = await kurtosisClient.getEnclaves();
+      const enclaves = Object.values(enclavesResponse.enclaveInfo);
+      const [starlarkRuns, services, filesAndArtifacts] = await Promise.all([
+        Promise.all(enclaves.map((enclave) => kurtosisClient.getStarlarkRun(enclave))),
```

Comment:
In the Cloud UI we've used `true-myth` for error handling. What do you think about adopting that for these API calls?

Reply:
That pattern seems reasonable. Generally I like to swallow application breaking errors as soon as possible, rather than propagating the handling to components that shouldn't need to know the real details of working with the api. So my plan was going to be:
* wrap the `kurtosisClient` in `useKurtosisClient` with some sort of `ErrorHandlingKurtosisClient`
* ErrorHandlingKurtosisClient would delegate all requests to the actual client used
* ErrorHandlingKurtosisClient, being in the react tree, would be able to create toasts with the error messages

Perhaps a combination of these makes it possible to have components render their own error state, but keeps the location of api errors consistent on the page.

Reply:
I like the idea of the `ErrorHandlingKurtosisClient`. What I care about is that we have some explicit notion of handling errors and making consumers of the api calls aware if an error has happened so they can deal with it.